firmware/arduino/chessboard/movementFunctions.ino

```diff
@@ -1,10 +1,58 @@
 bool moveDirect(int startCol, int startRow, int endCol, int endRow)
 {
-  // TODO:
-  // Must account for diagonal and straight movement cases
-  // Must also return true if the current position is the target position 
-  // (Since no change is needed)
-  return true;
+  uint8_t statusCodeResult;
+  uint8_t curCol, curRow;
+  int8_t deltaX, deltaY;
+  bool isSuccessful;
+
+  deltaX = endCol - startCol;
+  deltaY = endRow - startRow;
+
+  // If target point is equal to the start point
+  if (deltaX == 0  &&  deltaY == 0)
+    return true;
+
+  // Checks if the EM is aligned properly
+  if ((currPositionX % stepsPerUnitSpace) || (currPositionY % stepsPerUnitSpace))
+    return false;
+
+  curCol = currPositionX / (stepsPerUnitSpace * 8);
+  curRow = currPositionY / (stepsPerUnitSpace * 8);
+
+  // Make initial move to first position. Move diagonally since it's faster.
+  if (!moveDirect(curCol, curRow, startCol, startRow))
+    return false;
+
+  // Enable electromagnet
+  ledcWrite(EM_PWM_CHANNEL, PWM_HALF);
+
+  // Assume no errors, unless proven otherwise
```

Comment:
Can you update this comment to make it clear that we need this variable so that if movement fails, we still turn off the electromagnet before returning from the function?